src/disc/stk/Albany_ExtrudedSTKMeshStruct.cpp

```diff
@@ -195,6 +198,51 @@ Albany::ExtrudedSTKMeshStruct::ExtrudedSTKMeshStruct(const Teuchos::RCP<Teuchos:
   // Create a mesh specs object for EACH side set
   this->initializeSideSetMeshSpecs(comm);
 
+  // Get upper bound on lateral/upper workset sizes by using Ioss element counts on side blocks
+  if (worksetSizeMax == -1 && basalMeshSpec->singleWorksetSizeAllocation) {
+    // Set lateral workset sizes based on basal sidesets
+    for (auto bssName : basalMeshSpec->ssNames) {
+      // Get maximum workset size of basalside sideset
+      const auto& basalSideSetMeshSpecs = basalMeshSpec->sideSetMeshSpecs;
```

Comment:
I'm a bit confused. Are you trying to get the mesh specs of the side of the basal mesh, i.e., a 1d mesh? I wasn't aware we stored those. I thought we stored mesh specs only when we built a mesh, and I thought we didn't build a mesh for the side of the side mesh.

Reply:
When I was digging around the mesh specs it looked like there are two sidesets called lateralside. One is the 2d boundary of the fully extruded 3d mesh and the other is attached to basalside's mesh specs and appears to be the 1d boundary of the basalside mesh. So it looks like it gets stored but I don't know if it gets used at all by any evaluators since it has the same sideset name as lateralside.

Reply:
Yes, I'm using the workset size from the 1D sides of the basal mesh to set the lateral workset sizes. In the case that the basalside uses Ioss, we always stored the mesh specs of the 1D sides. I just added the correct workset sizes. I'm not exactly sure if the mesh is built on the side but I think it is since I see prints for STKDisc: sideset [1D sideset name]. I was also wondering if that was even correct...

Reply:
Mind=blown.

Reply:
Note sure it's what you are talking about, but we do that to give the possibility of having different lateral side sets, e.g. a lateral side where the ice is floating, one where it's grounded, one where we prescribe Dirichlet conditions. So you'll have sides like `extruded_boundary_side_set_1`, `extruded_boundary_side_set_2`, etc. We always have `lateralside` that contains all the lateral sides.

Reply:
What I'm saying is that I did not know that we built a mesh specs for the basal mesh side sets. I thought we only stored those sidesets as side sets, without the extra info coming with a mesh specs.

Reply:
Yeah, I just realized this too as I was working this. I imagine what happened was the general initialization of sidesets was put into Ioss for restart and 3D meshes but the consequence of that is that sideset mesh specs are constructed for any sideset that uses Ioss too.